operator/pkg/reconciliation/constructor.go

```diff
@@ -461,3 +291,169 @@ func calculatePodAntiAffinity(allowMultipleNodesPerWorker bool) *corev1.PodAntiA
+
+	// server logger container
+	loggerContainer := corev1.Container{}
+	loggerContainer.Name = "server-system-logger"
+	loggerContainer.Image = "busybox"
+	loggerContainer.Args = []string{
+		"/bin/sh", "-c", "tail -n+1 -F /var/log/cassandra/system.log",
+	}
+	loggerContainer.VolumeMounts = []corev1.VolumeMount{cassServerLogsMount}
+
+	return []corev1.Container{cassContainer, loggerContainer}, nil
+}
+
+func buildInitContainers(dc *api.CassandraDatacenter) ([]corev1.Container, error) {
+	serverCfg := corev1.Container{}
+	serverCfg.Name = "server-config-init"
+	serverCfg.Image = dc.GetConfigBuilderImage()
+	serverCfgMount := corev1.VolumeMount{
+		Name:      "server-config",
+		MountPath: "/config",
+	}
+	serverCfg.VolumeMounts = []corev1.VolumeMount{serverCfgMount}
+
+	configData, err := dc.GetConfigAsJSON()
+	if err != nil {
+		return nil, err
+	}
+	serverVersion := dc.Spec.ServerVersion
+	serverCfg.Env = []corev1.EnvVar{
+		{Name: "CONFIG_FILE_DATA", Value: configData},
+		{Name: "POD_IP", ValueFrom: selectorFromFieldPath("status.podIP")},
+		{Name: "RACK_NAME", ValueFrom: selectorFromFieldPath(fmt.Sprintf("metadata.labels['%s']", api.RackLabel))},
+		{Name: "PRODUCT_VERSION", Value: serverVersion},
+		{Name: "PRODUCT_NAME", Value: dc.Spec.ServerType},
+		// TODO remove this post 1.0
+		{Name: "DSE_VERSION", Value: serverVersion},
+	}
+
+	return []corev1.Container{serverCfg}, nil
+}
+
+func buildPodTemplateSpec(dc *api.CassandraDatacenter, zone string, rackName string) (*corev1.PodTemplateSpec, error) {
+	baseTemplate := dc.Spec.PodTemplate
+
+	if baseTemplate == nil {
+		baseTemplate = &corev1.PodTemplateSpec{}
+	}
+
+	podLabels := dc.GetRackLabels(rackName)
+	oplabels.AddManagedByLabel(podLabels)
+	podLabels[api.CassNodeState] = stateReadyToStart
+
+	baseTemplate.Labels = utils.MergeMap(baseTemplate.Labels, podLabels)
+
+	// affinity
+	affinity := &corev1.Affinity{}
+	affinity.NodeAffinity = calculateNodeAffinity(zone)
+	affinity.PodAntiAffinity = calculatePodAntiAffinity(dc.Spec.AllowMultipleNodesPerWorker)
+	baseTemplate.Spec.Affinity = affinity
+
+	// volumes
+	vServerConfig := corev1.Volume{}
+	vServerConfig.Name = "server-config"
+	vServerConfig.VolumeSource = corev1.VolumeSource{
+		EmptyDir: &corev1.EmptyDirVolumeSource{},
+	}
+
+	vServerLogs := corev1.Volume{}
+	vServerLogs.Name = "server-logs"
+	vServerLogs.VolumeSource = corev1.VolumeSource{
+		EmptyDir: &corev1.EmptyDirVolumeSource{},
+	}
+
+	volumes := []corev1.Volume{vServerConfig, vServerLogs}
+	baseTemplate.Spec.Volumes = append(baseTemplate.Spec.Volumes, volumes...)
+
+	serviceAccount := "default"
+	if dc.Spec.ServiceAccount != "" {
+		serviceAccount = dc.Spec.ServiceAccount
+	}
+	baseTemplate.Spec.ServiceAccountName = serviceAccount
+
+	// init containers
+	initContainers, err := buildInitContainers(dc)
+	if err != nil {
+		return nil, err
+	}
+	baseTemplate.Spec.InitContainers = append(baseTemplate.Spec.InitContainers, initContainers...)
```

Comment:
I think Jim wanted the init containers prepended to the ones the user provides, same with the containers.

Reply:
whoops... should be fixed now